ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -183,24 +183,31 @@ public Void visitLikePredicate(final LikePredicate node, final TypeContext conte
 
     @Override
     public Void visitFunctionCall(final FunctionCall node, final TypeContext context) {
-      final List<SqlArgument> argumentTypes = new ArrayList<>();
       final FunctionName functionName = node.getName();
+
+      final TypeContext contextCopy = context.getCopy();
+      final List<SqlArgument> argumentTypes = new ArrayList<>();
       final boolean hasLambda = node.hasLambdaFunctionCallArguments();
       for (final Expression argExpr : node.getArguments()) {
-        final TypeContext childContext = context.getCopy();
+        final TypeContext childContext;
```

Comment:
Better add some comments for other readers why we are doing this: since the context need to be shared across function args that involves lambdas, blah blah.